opentelemetry-api/src/opentelemetry/context/__init__.py

```diff
@@ -25,54 +23,35 @@
 from opentelemetry.util._importlib_metadata import entry_points
 
 logger = logging.getLogger(__name__)
-_RUNTIME_CONTEXT = None  # type: typing.Optional[_RuntimeContext]
-_RUNTIME_CONTEXT_LOCK = threading.Lock()
 
-_F = typing.TypeVar("_F", bound=typing.Callable[..., typing.Any])
 
-
-def _load_runtime_context(func: _F) -> _F:
-    """A decorator used to initialize the global RuntimeContext
+def _load_runtime_context() -> typing.Optional[_RuntimeContext]:
+    """Initialize the RuntimeContext
 
     Returns:
-        A wrapper of the decorated method.
+        An instance of RuntimeContext.
     """
 
-    @wraps(func)  # type: ignore[misc]
-    def wrapper(
-        *args: typing.Tuple[typing.Any, typing.Any],
-        **kwargs: typing.Dict[typing.Any, typing.Any],
-    ) -> typing.Optional[typing.Any]:
-        global _RUNTIME_CONTEXT  # pylint: disable=global-statement
-
-        with _RUNTIME_CONTEXT_LOCK:
-            if _RUNTIME_CONTEXT is None:
-                # FIXME use a better implementation of a configuration manager
-                # to avoid having to get configuration values straight from
-                # environment variables
-                default_context = "contextvars_context"
-
-                configured_context = environ.get(
-                    OTEL_PYTHON_CONTEXT, default_context
-                )  # type: str
-                try:
-
-                    _RUNTIME_CONTEXT = next(  # type: ignore
-                        iter(  # type: ignore
-                            entry_points(  # type: ignore
-                                group="opentelemetry_context",
-                                name=configured_context,
-                            )
-                        )
-                    ).load()()
-
-                except Exception:  # pylint: disable=broad-except
-                    logger.exception(
-                        "Failed to load context: %s", configured_context
-                    )
-        return func(*args, **kwargs)  # type: ignore[misc]
-
-    return typing.cast(_F, wrapper)  # type: ignore[misc]
+    # FIXME use a better implementation of a configuration manager
+    # to avoid having to get configuration values straight from
+    # environment variables
+    default_context = "contextvars_context"
+
+    configured_context = environ.get(
+        OTEL_PYTHON_CONTEXT, default_context
+    )  # type: str
+
+    return next(  # type: ignore
```

Comment:
Could you keep the exception handling as is?

Reply:
Would it be better to just throw the exception for failfast instead of catch it? Because the exception only happens when environment variable OTEL_PYTHON_CONTEXT set to unsupported value. It's also fine to failover to the default_context, however, it requires extra codes to load the default_context on exception catching.

Failfast is preffered, as I mentioned before https://github.com/open-telemetry/opentelemetry-python/pull/3763#discussion_r1549005382.

Reply:
Design-wise that might be the case but changing this would be a breaking change. I would suggest keeping it as is and maybe down the road we can explore user experience options via configuration.

Reply:
OK, it has been restored.